compiler/rustc_expand/src/mbe.rs

```diff
@@ -17,23 +17,52 @@ use rustc_data_structures::sync::Lrc;
 use rustc_span::symbol::Ident;
 use rustc_span::Span;
 
-/// Contains the sub-token-trees of a "delimited" token tree, such as the contents of `(`. Note
-/// that the delimiter itself might be `NoDelim`.
+/// Contains the sub-token-trees of a "delimited" token tree such as `($var)`. The delimiter itself
+/// might be `NoDelim`.
 #[derive(Clone, PartialEq, Encodable, Decodable, Debug)]
 struct Delimited {
     delim: token::DelimToken,
-    tts: Vec<TokenTree>,
+    /// Note: This contains the opening and closing delimiters tokens (e.g. `(` and `)`), *unless*
```

Comment:
FWIW, this is a pretty gross violation of the language model, since delimiters are not token trees.

But this is a pre-existing issue since `rustc` parser treats delimiters as regular tokens, if it's rewritten to not do that then the need to put delimiters into a common slice with regular tokens will disappear as well.